Content.Shared/Buckle/SharedBuckleSystem.Buckle.cs

```diff
@@ -484,6 +484,14 @@ private bool CanUnbuckle(Entity<BuckleComponent?> buckle, EntityUid? user, bool
         if (buckle.Comp.BuckledTo is not { } strapUid)
             return false;
 
+        if (user.HasValue && !HasComp<HandsComponent>(user))
+        {
+            // PopupPredicted when
+            if (_netManager.IsServer)
+                _popup.PopupEntity(Loc.GetString("buckle-component-no-hands-message"), user.Value, user.Value);
+            return false;
+        }
+
```

Comment:
As notafet said actionblocker is far more appropriate, also looks like you just copy-pasted this, as I wrote this comment a few years ago, but predicted popups have existed for years.